code/src/lidar.cpp

```diff
@@ -0,0 +1,91 @@
+#include <lidar.hpp>
+
+namespace r2d2::measuring_distance {
+    uint8_t lidar_c::receive_uint8() {
+        for (;;) {
+            if (uart.available() > 0) {
+                uint8_t byte = uart.receive();
+                checksum += byte;
+                return byte;
+            }
+        }
+    }
+
+    uint16_t lidar_c::receive_uint16() {
+        return (receive_uint8() << 8) | receive_uint8();
+    }
+
+    void lidar_c::wait_for_startbyte(uint8_t start_byte) {
+        for (;;) {
+            if (receive_uint8() == start_byte) {
+                checksum = start_byte;
+                break;
+            }
+        }
+    }
+
+    bool lidar_c::receive_packet_header() {
+        header.frame_length = receive_uint16();
+        header.protocol_version = receive_uint8();
+        header.frame_type = receive_uint8();
+        header.command_word = receive_uint8();
+        header.data_length = receive_uint16();
+        header.radar_speed = receive_uint8();
+
+        if (header.frame_type != 0x61) {
```

Comment:
I saw the 0x61 in a comment somewhere, but please extract this to a named constant